### PR TITLE
Add geant4-data-channeling pkg needed by Geant4 11.3.0

### PR DESCRIPTION
**Pull request by tkittel**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by tkittel** [comment]
To clarify, this package is needed for the geant4 package, since in release 11.3.0 it introduced a new kind of dataset. Since whomever created the geant4 packages originally decided that each dataset goes in their own feedstock, I am simply continuing the tradition here (I am simply trying to keep the geant4 packages alive since they are so important, but I don't have the cycles to give everything a complete overhaul).

The recipe is more or less copy paste from the geant4-data-particlexs feedstock.

**Comment by tkittel** [comment]
Note that this is a noarch package with some bash commands in the meta.yaml ruining the windows build. However, this does not actually ruin the package (and in any case the geant4-feedstock is non-windows).

**Comment by tkittel** [comment]
@conda-forge/staged-recipes, ready for review!

**Comment by tkittel** [comment]
Dear bot. This is a noarch package which merely unpacks a tarball, so I could not find a more specific team.

File: recipes/geant4-data-channeling/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+{% set name = "geant4-data-"+dataset_name.lower() %}
+{% set install_dir = "share/Geant4/data/"+dataset_name+version_canonical_geant4 %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://geant4-data.web.cern.ch/geant4-data/datasets/G4{{ dataset_name }}.{{ version_canonical_geant4 }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (xhochy) [inline recipes/geant4-data-channeling/meta.yaml:17]: Less Jinja indirection
```suggestion
  sha256: 203e3c69984ca09acd181a1d31a9b0efafad4bc12e6c608f0b05e695120d67f2
```

File: recipes/geant4-data-channeling/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{% set dataset_name = "CHANNELING" %}
+{% set version = "1.0.0" %}
+{% set version_canonical_geant4 = "1.0" %}
+{% set sha256 = "203e3c69984ca09acd181a1d31a9b0efafad4bc12e6c608f0b05e695120d67f2" %}
```
Comment (xhochy) [inline recipes/geant4-data-channeling/meta.yaml:4]:
```suggestion
```

**Review by xhochy — COMMENTED** [review]
Small nit, otherwise this is good to go.

**Comment by tkittel** [comment]
Thanks for the quick review @xhochy ! I committed your suggestions and did a rebase on main for good measure. So hopefully good to go now :-)

File: recipes/geant4-data-channeling/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    - make
+
+test:
+  commands:
+    - '[ "${{ env_var }}" == "$CONDA_PREFIX/{{ install_dir }}" ]'
+
+about:
+  home: https://geant4.web.cern.ch/
+  summary: {{ summary }}
+  license: OTHER
```
Comment (xhochy) [inline recipes/geant4-data-channeling/meta.yaml:40]: Actually, we still need to attach the correct license. OTHER is a bit unspecific.
```suggestion
  license: LicenseRef-Geant4
```

**Comment by tkittel** [comment]
@xhochy yeah I agree. I wish Geant4 would have picked a standard open source license, but your change is clearly better.

**Review by xhochy — APPROVED** [review]

**Pull request merged**